src/main/proto/wfa/measurement/reporting/v1alpha/metric_set.proto

```diff
@@ -0,0 +1,109 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package wfa.measurement.reporting.v1alpha;
+
+import "google/api/field_behavior.proto";
+import "google/api/resource.proto";
+import "wfa/measurement/reporting/v1alpha/time_interval.proto";
+
+option java_package = "org.wfanet.measurement.reporting.v1alpha";
+option java_multiple_files = true;
+
+// A metric set using data from the Cross-Media Measurement System.
+message MetricSet {
+  string measurement_consumer = 1 [(google.api.resource_reference) = {
+    type: "halo.wfanet.org/MeasurementConsumer"
+  }];
+
+  message EventGroupFilter {
+    // Key of the map entry, which is an `EventGroup` resource name. Required.
+    string key = 1 [
+      (google.api.field_behavior) = REQUIRED,
+      (google.api.resource_reference) = {
+        type: "reporting.halo-cmm.org/EventGroup"
+      }
+    ];
+
+    // Filter predicate in CEL.
+    string value = 2 [(google.api.field_behavior) = REQUIRED];
+  }
+
+  // Map of EventGroup to filter predicate.
+  repeated EventGroupFilter event_group_filters = 2
+      [(google.api.field_behavior) = REQUIRED];
+
+  enum Metric {
+    METRIC_UNSPECIFIED = 0;
+    REACH = 1;
+    FREQUENCY_HISTOGRAM = 2;
+    REACH_CURVE = 3;
+    IMPRESSION_COUNT = 4;
+    WATCH_DURATION = 5;
+  }
+  repeated Metric metrics = 3 [(google.api.field_behavior) = REQUIRED];
+
+  oneof time {
+    TimeInterval time_interval = 4;
+    RepeatedTimeInterval repeated_time_interval = 5;
+  }
+
+  // Represents a binary set operation.
+  message SetOperation {
+    enum Type {
+      TYPE_UNSPECIFIED = 0;
+      UNION = 1;
+      DIFFERENCE = 2;
+      INTERSECTION = 3;
+    }
+    Type type = 1 [(google.api.field_behavior) = REQUIRED];
+
+    message SetSpec {
+      // The name of the set for display purposes.
+      //
+      // e.g. this may be a column or row header in CSV output.
+      string display_name = 1 [(google.api.field_behavior) = REQUIRED];
+
+      // Set of EventGroup resource names.
+      repeated string event_groups = 2 [
+        (google.api.resource_reference).type =
+            "reporting.halo-cmm.org/EventGroup",
+        (google.api.field_behavior) = REQUIRED
+      ];
+
+      // CEL filter predicate that applies to all `event_groups`.
+      //
+      // This filter and the one in the `event_group_filters` form a conjunction
+      // that specifies which impressions are included in this set. If
+      // unspecified, evaluates to `true`.
+      string filter = 3;
```

Comment:
I wonder if this should live at the MetricSet level instead, or initially, we could add a lower level one later.